pkg/visualizer2/vis_graph.go

```diff
@@ -0,0 +1,53 @@
+package visualizer
+
+import (
+	"errors"
+
+	"github.com/dominikbraun/graph"
+	construct "github.com/klothoplatform/klotho/pkg/construct2"
+	"github.com/klothoplatform/klotho/pkg/graph_addons"
+	"github.com/klothoplatform/klotho/pkg/set"
+)
+
+type (
+	VisResource struct {
+		ID construct.ResourceId
+
+		Tag      string
+		Parent   construct.ResourceId
+		Children set.Set[construct.ResourceId]
+	}
+
+	VisGraph graph.Graph[construct.ResourceId, *VisResource]
+)
```

Comment:
what are these used for?

Reply:
nevermind i think i see were using a graph instead of the maps that were being used in the output function

Reply:
`VisGraph` is basically what `Topology` used to be, just moved into visualizer package and using the `graph` library instead of our own Nodes & Edges struct.